### PR TITLE
add metric for repair request blockstore misses

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
There are no metrics to determine whether shred data for repair requests have been found in blockstore.

#### Summary of Changes
Add metric to track unsuccessful blockstore queries from repair requests.

Fixes #

### CODE REVIEW

**Comment by bw-solana** [comment]
Looks like there are quite a few things that can lead to empty batch being returned. `blockstore_misses` sounds sufficiently vague that it could lead to confusion. Maybe add a comment to `blockstore_misses` in the `ServeRepairStats` struct to indicate what it covers?

**Comment by jbiseda** [comment]
Updated this to only track window index misses as this is the request type where the requester is specifying a slot/index explicitly.

File: core/src/serve_repair.rs [hunk 1]
```diff
@@ -1214,8 +1217,6 @@ impl ServeRepair {
             from_addr,
             nonce,
         )?;
-
-        inc_new_counter_debug!("serve_repair-window-request-ledger", 1);
```
Comment (bw-solana) [inline core/src/serve_repair.rs:1218]: Did you mean to remove this?

**Comment by bw-solana** [comment]
> Updated this to only track window index misses as this is the request type where the requester is specifying a slot/index explicitly.

I thought `repair_response_packet` guaranteed we got the data shred because of this line: `.expect("Blockstore could not get data shred");`

Looks like we can return None here for quite a few cases, including shred data exceeding packet size.

File: core/src/serve_repair.rs (hunk 1 above)
Comment (jbiseda) [reply core/src/serve_repair.rs:1218]: Yes, it would be tracking the inverse of the new metric.

**Comment by jbiseda** [comment]
> I thought `repair_response_packet` guaranteed we got the data shred because of this line: `.expect("Blockstore could not get data shred");`

This expect should be operating on the `Result` wrapping the `Option<Vec<u8>>`.
 
> Looks like we can return None here for quite a few cases, including shred data exceeding packet size.

I think exceeding packet size would be the only case here (other than just not having that particular shred in blockstore), but the shred data should be constrained by `shred.bytes_to_store()` during insertion. This metric is counting the number of "misses" from `run_window_request`, so it would count blockstore shred data that couldn't be returned as a "miss".

**Review by bw-solana — APPROVED** [review]
LGTM

**Pull request merged**